### PR TITLE
feat(maven): Skip lookup for parent packages

### LINKED ISSUE #18117: Dependency Lookup Warnings for Internal Maven Artifacts
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

I have a public GitHub repository on which I use Maven. Surprisingly, Renovate adds dependency lookup warnings for internal Maven artifacts to the dependency dashboard. All these warnings are false positives. For example, `org.tinylog:parent` is my parent POM and can be found in the `pom.xml` in the root directory of my repository.

Dependency Dashboard: https://github.com/tinylog-org/tinylog/issues/296
Renovate Configuration: https://github.com/tinylog-org/tinylog/blob/v2.6/.github/renovate.json
Base Branch: https://github.com/tinylog-org/tinylog/tree/v3.0

I have noticed one peculiarity: `v2.6` is my default branch, but `v3.0` is my base branch. If I fork my repository and make `v3.0` there to the default branch, the bug will not occur. In this case, there are no such false dependency warnings.

### Relevant debug logs

<details><summary>Logs</summary>

https://app.renovatebot.com/dashboard#github/tinylog-org/tinylog/838133334

```
INFO: Repository started
{
  "renovateVersion": "32.214.0"
}
DEBUG: Using localDir: /mnt/renovate/gh/tinylog-org/tinylog
DEBUG: PackageFiles.clear() - Package files deleted
{
  "baseBranches": []
}
DEBUG: initRepo("tinylog-org/tinylog")
DEBUG: Using queue
{
  "concurrency": 10,
  "host": "api.github.com"
}
DEBUG: tinylog-org/tinylog default branch = v2.6
DEBUG: Using app token for git init
DEBUG: RepoCacheBase.load() - expecting data of type 'string' received 'object' instead - skipping
DEBUG: Resetting npmrc
DEBUG: detectSemanticCommits()
DEBUG: Initializing git repository into /mnt/renovate/gh/tinylog-org/tinylog
DEBUG: Performing blobless clone
DEBUG: git clone completed
{
  "durationMs": 1855
}
EBUG: getCommitMessages
DEBUG: Semantic commits detection: unknown
DEBUG: No semantic commits detected
DEBUG: checkOnboarding()
DEBUG: isOnboarded()
DEBUG: findFile(renovate.json)
DEBUG: findFile(renovate.json5)
DEBUG: findFile(.github/renovate.json)
DEBUG: Config file exists
{
  "fileName": ".github/renovate.json"
}
DEBUG: Retrieving issueList
DEBUG: Retrieved 0 issues
DEBUG: Repo is onboarded
DEBUG: Found .github/renovate.json config file
DEBUG: Repository config
{
  "fileName": ".github/renovate.json",
  "config": {
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "baseBranches": [
      "v3.0"
    ]
  }
}
DEBUG: migrateAndValidate()
DEBUG: No config migration necessary
DEBUG: massaged config
{
  "config": {
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "baseBranches": [
      "v3.0"
    ],
    "extends": [
      "github>whitesource/merge-confidence:beta"
    ]
  }
}
DEBUG: migrated config
{
  "config": {
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "baseBranches": [
      "v3.0"
    ],
    "extends": [
      "github>whitesource/merge-confidence:beta"
    ]
  }
}
DEBUG: Looking up org.tinylog:parent in repository https://repo.maven.apache.org/maven2/
{
  "baseBranch": "v3.0"
}
DEBUG: Content is not found for Maven url
{
  "baseBranch": "v3.0",
  "url": "https://repo.maven.apache.org/maven2/org/tinylog/parent/maven-metadata.xml"
}
DEBUG: Failed to look up dependency org.tinylog:parent (org.tinylog:parent)(packageFile="benchmarks/pom.xml", dependency="org.tinylog:parent")
{
  "baseBranch": "v3.0"
}
DEBUG: Failed to look up dependency org.tinylog:parent (org.tinylog:parent)(packageFile="slf4j-tinylog/pom.xml", dependency="org.tinylog:parent")
{
  "baseBranch": "v3.0"
}
DEBUG: Failed to look up dependency org.tinylog:parent (org.tinylog:parent)(packageFile="tinylog-api-kotlin/pom.xml", dependency="org.tinylog:parent")
{
  "baseBranch": "v3.0"
}
DEBUG: Failed to look up dependency org.tinylog:parent (org.tinylog:parent)(packageFile="tinylog-api/pom.xml", dependency="org.tinylog:parent")
{
  "baseBranch": "v3.0"
}
DEBUG: Failed to look up dependency org.tinylog:parent (org.tinylog:parent)(packageFile="tinylog-core-base/pom.xml", dependency="org.tinylog:parent")
{
  "baseBranch": "v3.0"
}
DEBUG: Failed to look up dependency org.tinylog:parent (org.tinylog:parent)(packageFile="tinylog-core-modern/pom.xml", dependency="org.tinylog:parent")
{
  "baseBranch": "v3.0"
}
DEBUG: Failed to look up dependency org.tinylog:parent (org.tinylog:parent)(packageFile="tinylog-core/pom.xml", dependency="org.tinylog:parent")
{
  "baseBranch": "v3.0"
}
DEBUG: Failed to look up dependency org.tinylog:parent (org.tinylog:parent)(packageFile="tinylog-impl/pom.xml", dependency="org.tinylog:parent")
{
  "baseBranch": "v3.0"
}
```

</details>

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

When package described in `pom.xml` has been referenced as top-level parent, skip lookups for every dependency referencing it.

## Context

- Closes: #18117

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
Would there ever be a reason why someone would want to upgrade a parent Pom? If so then we should think to make it possible to override using package rules

**Comment by zharinov** [comment]
Probably what we can do is to mark these `depType`s as `root-parent` and create default host rule preset that ignores it?

File: lib/workers/repository/errors-warnings.ts [hunk 1]
```diff
@@ -111,7 +112,9 @@ export function getDepWarningsDashboard(
   }
 
   const depWarnings = warnings
-    .map((w) => w.replace('Failed to look up dependency ', ''))
+    .map((w) =>
+      w.replace(regEx(/^Failed to look up(?: [-\w]+)? dependency /), '')
+    )
```
Comment (zharinov) [inline lib/workers/repository/errors-warnings.ts:117]: Would be made a separate PR, but meh

Comment (viceice) [reply lib/workers/repository/errors-warnings.ts:117]: yes, please do this in a seperate PR, it's unrelated

**Review by viceice — COMMENTED** [review]
otherwise lgtm

@rarkins Should be marked as feat instead of fix?

File: lib/workers/repository/errors-warnings.ts (hunk 1 above)
Comment (zharinov) [inline lib/workers/repository/errors-warnings.ts:117]:
```suggestion
    .map((w) => w.replace('Failed to look up dependency ', ''))
```

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]
